cmd/snap/cmd_advise.go

```diff
@@ -205,11 +210,50 @@ func adviseViaAptHook() error {
 	return nil
 }
 
+type Snap struct {
+	Snap    string
+	Version string
+	Command string
+}
+
+func dumpDbHook() error {
+	commands, err := advisor.DumpCommands()
+	if err != nil {
+		return err
+	}
+
+	commands_processed := make([]string, 0)
+	var b []Snap
+
+	for key, value := range commands {
+		err := json.Unmarshal([]byte(value), &b)
+		if err != nil {
+			return err
+		}
+		for i := range b {
+			var s = fmt.Sprintf("%s\xff%s/s\n", key, b[i].Snap)
```

Comment:
I'm curious about the format, I assume this is specific to your tool? Would you mind if we make it a little bit more generic and you did a tiny bit of processing when importing the data? Like: `s := fmt.Sprintf("%s:%s\n")` ? This should be fine parsing wise, the ":" is not legal as part of a snap name so you can always split on the last ":" to get the (command,snap) pair. Alternatively we can keep the \ff but I would like to avoid the extra "/s" which seems to indicate snap(?).

Reply:
> which seems to indicate snap(?)
correct. u is universe, n non-free, et cetera.

Yeah I don't mind if you use a slightly different format. I used \xff because it is not a valid UTF-8 byte. (along with \xfe and 6 others if you limit UTF-8 to the unicode range specified in  RFC 3629) As long as it isn't so fancy that I can't use crappy sed to fix it back to what I have it as.

Reply:
Now I remember. I used \xff because I wanted coreutils' "sort" to work correctly on the data.

Reply:
Thanks, I think \xff is fine. I would like to avoid the suffix though in case other tools want to use the db export for which the /s will not be that useful. I assume you could easily add it when you import the data, right?

Reply:
Of course